### PR TITLE
add recommendation doc

### PR DESCRIPTION
**Pull request by songhappy**

### CODE REVIEW

File: docs/docs/models/recommendation/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Analytics Zoo Object Detection API
```
Comment (hkvision) [inline docs/docs/models/recommendation/README.md]: Object Detection -> Recommendation

File: docs/docs/models/recommendation/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[Scala example](https://github.com/intel-analytics/zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/NeuralCFexample.scala)
+
+Users can build a NCF model using below code piece. 
+
+```scala
+//set up feature column information for feature engineering and model
+val ncf = NeuralCF[Float](
+      userCount = userCount,
+      itemCount = itemCount,
+      numClasses = params.numClasses ,
```
Comment (hkvision) [inline docs/docs/models/recommendation/README.md]: removed these undefined values? Add the default values.

File: docs/docs/models/recommendation/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+WND Learning Model, proposed by ([Google, 2016](https://arxiv.org/pdf/1606.07792.pdf)), is a DNN-Linear mixed model. WND combines the strength of memorization and generalization. It's useful for generic large-scale regression and classification problems with sparse input features(e.g., categorical features with a large number of possible feature values). It has been used for Google App Store for their app recommendation.
+[Scala example](https://github.com/intel-analytics/zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/WideAndDeepExample.scala)
+
+It's very easy to build a WND model for recommendation with below code piece. 
+
+```scala
+//set up feature column information for feature engineering and model
+val localColumnInfo = ColumnFeatureInfo("...")
+val wideAndDeep: WideAndDeep[Float] = WideAndDeep[Float](
+      params.modelType,
```
Comment (hkvision) [inline docs/docs/models/recommendation/README.md]: specify what model types are supported?

**Comment by hkvision** [comment]
Thanks for the documentation. I will add the python part after this is merged.

**Comment by jason-dai** [comment]
Is this ready for merge?

**Pull request merged**